src/crypto/ed25519.js

```diff
@@ -0,0 +1,33 @@
+'use strict'
+
+const nacl = require('tweetnacl')
+
+exports.publicKeyLength = nacl.sign.publicKeyLength
+exports.privateKeyLength = nacl.sign.secretKeyLength
+
+exports.generateKey = function (callback) {
+  const done = (err, res) => setImmediate(() => {
```

Comment:
Please use `setImmediate = require('async/setImmediate')` to ensure we have setImmediate actually available. (yes this is not yet consistently fixed in the code base :( )